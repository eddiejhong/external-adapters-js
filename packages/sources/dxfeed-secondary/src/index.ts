import { expose } from '@chainlink/ea-bootstrap'
import * as dxfeed from '@chainlink/dxfeed-adapter'
import { makeConfig, NAME } from './config'

export = {
  NAME,
  makeExecute: dxfeed.makeExecute,
  makeConfig,
  ...expose(NAME, dxfeed.makeExecute(), dxfeed.makeWSHandler(), dxfeed.endpointSelector),
}
