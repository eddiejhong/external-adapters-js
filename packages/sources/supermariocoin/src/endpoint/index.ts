export * as highlow from './highlow'
export * as total from './total'
