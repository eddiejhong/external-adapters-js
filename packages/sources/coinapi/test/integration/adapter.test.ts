import { AdapterRequest } from '@chainlink/types'
import http from 'http'
import nock from 'nock'
import request from 'supertest'
import { server as startServer } from '../../src/index'
import { mockCryptoEndpointFailure, mockCryptoEndpointSuccess } from './cryptoFixtures'

let oldEnv: NodeJS.ProcessEnv

describe('coinmarketcap', () => {
  let server: http.Server
  const req = request('localhost:8080')

  beforeAll(async () => {
    oldEnv = JSON.parse(JSON.stringify(process.env))
    process.env.CACHE_ENABLED = 'false'
    process.env.API_KEY = process.env.API_KEY || 'mock-api-key'
    if (process.env.RECORD) {
      nock.recorder.rec()
    }
    server = await startServer()
  })
  afterAll((done) => {
    process.env = oldEnv
    if (process.env.RECORD) {
      nock.recorder.play()
    }

    nock.restore()
    nock.cleanAll()
    nock.enableNetConnect()
    server.close(done)
  })

  describe('coinapi crypto endpoint', () => {
    describe('when sending well-formed request', () => {
      it('should reply with success', async () => {
        const cryptoRequest: AdapterRequest = {
          id: '1',
          data: {
            endpoint: 'crypto',
            base: 'ETH',
            quote: 'BTC',
          },
        }
        mockCryptoEndpointSuccess()
        const response = await req
          .post('/')
          .send(cryptoRequest)
          .set('Accept', '*/*')
          .set('Content-Type', 'application/json')
          .expect('Content-Type', /json/)
          .expect(200)
        expect(response.body).toMatchSnapshot()
      })
    })

    describe('when sending request without quote asset', () => {
      it('should reply with failure', async () => {
        const cryptoRequest: AdapterRequest = {
          id: '1',
          data: {
            endpoint: 'crypto',
            base: 'ETH',
          },
        }
        mockCryptoEndpointFailure()
        const response = await req
          .post('/')
          .send(cryptoRequest)
          .set('Accept', '*/*')
          .set('Content-Type', 'application/json')
          .expect('Content-Type', /json/)
          .expect(400)
        expect(response.body).toMatchSnapshot()
      })
    })
  })
})
