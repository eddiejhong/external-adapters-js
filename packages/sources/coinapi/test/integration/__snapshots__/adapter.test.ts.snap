// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coinmarketcap coinapi crypto endpoint when sending request without quote asset should reply with failure 1`] = `
Object {
  "error": Object {
    "message": "Required parameter not supplied: quote",
    "name": "AdapterError",
  },
  "jobRunID": "1",
  "status": "errored",
  "statusCode": 400,
}
`;

exports[`coinmarketcap coinapi crypto endpoint when sending well-formed request should reply with success 1`] = `
Object {
  "data": Object {
    "result": 0.06262119731705901,
  },
  "jobRunID": "1",
  "result": 0.06262119731705901,
  "statusCode": 200,
}
`;
